spek-runtime/jvm/src/main/kotlin/org/spekframework/spek2/runtime/SpekJvmRuntime.kt

```diff
@@ -38,15 +38,13 @@ actual class SpekRuntime: AbstractRuntime() {
             .filter { !it.isAbstract }
             .map { klass ->
                 klass to PathBuilder.from(klass).build()
+            }.map { (klass, path) ->
+                val matched = discoveryRequest.paths.firstOrNull { it.isRelated(path) }
+                matched to resolveSpec(instanceFactoryFor(klass).create(klass), path)
```

Comment:
I think that there is some redundancy here. Even if path was not matched you are calling `resolveSpec` for it.

Reply:
👍